### PR TITLE
fix: showPublicKey bug with address import

### PR DESCRIPTION
**Pull request by JeremiGendron**
## Proposed changes

#771 doesn't set the activeWallet, which means if you switch to a second wallet the fix is voided. This is fixed in this PR and refactored a little bit, so as to store the id of the wallet only and not the whole object.

The other change I've implemented here is to watch for changes in the currentWallet's publicKey in the `WalletSidebar` in order to check whether the wallet has one. If the wallet has no publicKey, it can cause a bug where the heading doesn't get reset.

1. The wallet heading reset is still misbehaving
 - Before
before_heading_noset

 - After
after_heading_set

2. The publicKey bug mentioned above (imported wallet with address only)
 - Before
before_publickey_noreset

 - After
after_publickey_reset

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (improve a current implementation without adding a new feature or fixing a bug)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build (changes that affect the build system)
- [ ] Docs (documentation only changes)
- [ ] Test (adding missing tests or fixing existing tests)
- [ ] Other... Please describe:

## Checklist

- [x] I have read the [CONTRIBUTING](https://docs.ark.io/guidebook/contribution-guidelines/contributing.html) documentation
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by ItsANameToo** [comment]
Activewallet is set in `resetHeading()` tho: https://github.com/ArkEcosystem/desktop-wallet/blob/8cbdcb0268f7d8385a4ebbbcad4999de8c35e408/src/renderer/components/Wallet/WalletHeading/WalletHeading.vue#L55

**Comment by ItsANameToo** [comment]
Also: I'm unable to reproduce what you show in your gif. With latest develop I can't get the heading to reset itself at random 🤷‍♂️

**Comment by JeremiGendron** [comment]
@ItsANameToo I admittedly didn't review the changes that well, let me see if I can reproduce in develop.

File: src/renderer/components/Wallet/WalletHeading/WalletHeading.vue [hunk 1]
```diff
@@ -39,7 +39,7 @@ export default {
 
   watch: {
     currentWallet () {
-      if (this.activeWallet && this.activeWallet.id !== this.currentWallet.id) {
+      if (this.activeWallet !== this.currentWallet.id) {
```
Comment (ItsANameToo) [inline src/renderer/components/Wallet/WalletHeading/WalletHeading.vue]: You should add `&& this.activeWallet.id` again, as it currently doesn't check against an id anymore

Comment (JeremiGendron) [reply src/renderer/components/Wallet/WalletHeading/WalletHeading.vue]: cool, on it!

**Pull request merged**